### PR TITLE
[Feature] Update backend gem section

### PR DESCRIPTION
**Pull request by ruslantolstov**

### CODE REVIEW

File: platform/backend/backend.md [hunk 1]
```diff
@@ -54,27 +78,154 @@ We don’t use Windows as OS for development machines. We use latest macOS or Ub
...
+
+* [pundit](https://github.com/elabs/pundit) - Minimal authorization through OO design and pure Ruby classes.
+* [rolify](https://github.com/RolifyCommunity/rolify) - Role management library with resource scoping.
+
+### Active Record
+
+* [AASM](https://github.com/aasm/aasm) - State machines for Ruby classes.
+* [active_record_union](https://github.com/brianhempel/active_record_union) - UNIONs in ActiveRecord! Adds proper union and union_all methods to ActiveRecord::Relation.
+* [PaperTrail](https://github.com/paper-trail-gem/paper_trail) - PaperTrail lets you track changes to your models' data. It's good for auditing or versioning.
+* [paranoia](https://github.com/rubysherpas/paranoia) - ActiveRecord plugin allowing you to hide and restore records without actually deleting them.
```
Comment (TeTiRoss) [inline platform/backend/backend.md]: > README:
Notice:
paranoia has some surprising behaviour (like overriding ActiveRecord's delete and destroy) and is not recommended for new projects. See discard's README for more details.
Paranoia will continue to accept bug fixes and support new versions of Rails but isn't accepting new features.

Comment (ruslantolstov) [reply platform/backend/backend.md]: @TeTiRoss, Agree 👍

File: platform/backend/backend.md [hunk 2]
```diff
@@ -54,27 +78,154 @@ We don’t use Windows as OS for development machines. We use latest macOS or Ub
...
+    * [image_processing](https://github.com/janko/image_processing) - High-level image processing helper methods with libvips and ImageMagick/GraphicsMagick.
+    * [shrine-url](https://github.com/shrinerb/shrine-url) - Custom URL storage for Shrine.
+
+## HTTP Clients
+
+* [Faraday](https://github.com/lostisland/faraday) - HTTP client lib that provides a common interface over many adapters (such as Net::HTTP) and embraces the concept of Rack middleware when processing the request/response cycle.
+
+## Searching
+
+* [searchkick](https://github.com/ankane/searchkick) - Intelligent search made easy with Rails and Elasticsearch.
```
Comment (TeTiRoss) [inline platform/backend/backend.md]: What about https://github.com/elastic/elasticsearch-rails? It is listed on Elasticsearch Rails integration page: https://www.elastic.co/guide/en/elasticsearch/client/ruby-api/current/ruby_on_rails.html.

Comment (ruslantolstov) [reply platform/backend/backend.md]: @TeTiRoss, Ok. It's more common solution.

**Pull request merged**